### DISCUSSION TITLE
Populate search when opening ListRecords

### DISCUSSION THREAD

**Discussion by MagicDidou** [filamentphp/filament#12141] [Help] [answered]
### Package

Panel builder

### Package Version

v3.x

### How can we help you?

Hi everybody,

I'm trying to do something but not sure if this is possible. Let's make it simple.

I have 2 tables **Customer** and **Invoice** with relationship (Invoice belongs to Customer and Customer has many Invoices). On the table view of **Customer** I have added an action button to jump to the Invoice list records view :
```
Action::make('see_invoices')
->url(fn (Customer $record): string => urlencode('invoices'......))
```
which I can customize with record data (e.g. parameters of the request) and I would like the **Invoice** view, which contains a **Customer** (`customers.name`) column, to be displayed with a search on Customer name already populated with the customer used to jump there.

I haven't seen any "Hook" on ListRecords page (such as the ones on Create/Edit) that could be used to do that.

Any help will be appreciated
Didier

**Comment by leandrocfe** [comment, marked as answer]
Add a filter in the InvoiceResource table
```php
->filters([
    SelectFilter::make('customer')
        ->relationship('customer', 'name'),
])
```

CustomerResource
```php
Action::make('see invoices')
    ->url(fn ($record): string => InvoiceResource::getUrl('index', ['tableFilters[customer][value]' => $record->id]))
```

**Reply by MagicDidou** [reply]
Thanks a lot. I will try this as soon as I can. Will let you know.

**Reply by MagicDidou** [reply]
It works perfectly ! Thanks a lot !
